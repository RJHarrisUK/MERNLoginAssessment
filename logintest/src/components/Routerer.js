import React, { Component } from "react";
import axios from "axios";
import { BrowserRouter as Router, Route } from "react-router-dom";

import Navigationbar from './Navigationbar.js';
import RegisterUser from './RegisterUser.js';
import LoginUser from './LoginUser.js';
import UserList from './UserList.js';

export default class Routerer extends Component {

    constructor() {
        super();
        this.state = {
            data: []
        }
    }

    componentDidMount() {
        this.getAll();
    }

    getAll = () => {
        axios
            .get('http://localhost:5000/User/all')
            .then(response => {

                this.setState({
                    data: response.data
                })
                console.log(response.data);
            })
    }

    render() {
        return (
            <div>
                <Router>
                    <Navigationbar />
                    <Route exact path="/" render={() => <RegisterUser onload={this.onLoad} />} />
                    <Route path="/Login" render={() => <LoginUser data={this.state.data} />} />
                    <Route path="/List" render={() => <UserList data={this.state.data} />} />
                </Router>
            </div>
        )
    }

}