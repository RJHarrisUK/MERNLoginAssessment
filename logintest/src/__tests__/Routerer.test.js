import React from "react";
import { MemoryRouter } from "react-router-dom";
import renderer from 'react-test-renderer';
import Routerer from '../components/Routerer.js';

it("render the router comp", () => {

    const wrapper = renderer.create(
        <MemoryRouter>

            <Routerer />

        </MemoryRouter>
    ).toJSON();

    expect(wrapper).toMatchSnapshot();

}); 