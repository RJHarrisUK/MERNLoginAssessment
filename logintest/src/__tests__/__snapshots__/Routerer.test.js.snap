// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the router comp 1`] = `
<div>
  <div>
    <nav
      className="navbar navbar-expand-md navbar-dark bg-dark"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        Register User
      </a>
      <button
        className="navbar-toggler"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        style={Object {}}
      >
        <ul
          className="ml-auto navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/Login"
              onClick={[Function]}
            >
              Login User
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/List"
              onClick={[Function]}
            >
              Show Users
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
  <div>
    <h1>
      Register User
    </h1>
    <form
      onSubmit={[Function]}
    >
      <input
        id="username"
        placeholder="Username"
        type="text"
      />
      <p
        style={
          Object {
            "color": "red",
          }
        }
      >
        
      </p>
      <br />
      <input
        id="email"
        placeholder="E-mail"
        type="text"
      />
      <p
        style={
          Object {
            "color": "red",
          }
        }
      >
        
      </p>
      <br />
      <input
        id="password"
        placeholder="Password"
        type="text"
      />
      <p
        style={
          Object {
            "color": "red",
          }
        }
      >
        
      </p>
      <br />
      <input
        id="password2"
        placeholder="Re-type Password"
        type="text"
      />
      <p
        style={
          Object {
            "color": "red",
          }
        }
      >
        
      </p>
      <br />
      <button
        type="submit"
      >
        Register User
      </button>
    </form>
  </div>
</div>
`;
